### DISCUSSION TITLE
Unexpected output with using colspan

### DISCUSSION THREAD

**Discussion by chiu0602** [tailwindlabs/tailwindcss#11947] [Help] [answered]
The layout below works expected.

```html
<div class="grid-cols-1 grid">
  <div class="col-span-1">col 1</div>
  <div class="col-span-3">col 2</div>
  <div class="col-span-2">col 3</div>
  <div class="col-span-2">col 4</div>
</div>
```

It looks like this:
|    |
|---|
| col1 |
| col2 |
| col3 |
| col4 |

[https://play.tailwindcss.com/AKuvmO7Vod](https://play.tailwindcss.com/AKuvmO7Vod)

---

However, when `col 5` is added, the layout works unexpected.

```html
<div class="grid-cols-1 grid">
  <div class="col-span-1">col 1</div>
  <div class="col-span-3">col 2</div>
  <div class="col-span-2">col 3</div>
  <div class="col-span-2">col 4</div>
  <div class="col-span-4">col 5</div>
</div>
```

It looks like this:
|   |   |
|---|---|
| col1 | col2 |
| col3 | col4 |
| col5 |

[https://play.tailwindcss.com/0M9DTSUmMq](https://play.tailwindcss.com/0M9DTSUmMq)

Anyone knows why and whether it can be avoided? Thanks.

**Comment by wongjn** [comment, marked as answer]
It goes two column because the `grid-columns: span 4 / span 4` applied via the `col-span-4` class forces the grid layout to be 4 column. Thus, there is sufficient columns for the `col-span-1` and `col-span-3` to be side-by-side, and the same for the `col-span-2` elements.

To avoid this, you could consider:
- Removing the grid layout altogether. The elements would naturally display in one column:
  ```html
  <div>
    <div>col 1</div>
    <div>col 2</div>
    <div>col 3</div>
    <div>col 4</div>
    <div>col 5</div>
  </div>
  ```
  [Tailwind Play example](https://play.tailwindcss.com/xuEkKe41ip)
- Applying `grid-column-start: 1` to all elements:
  ```html
  <div class="grid-cols-1 grid">
    <div class="col-start-1 col-span-1">col 1</div>
    <div class="col-start-1 col-span-3">col 2</div>
    <div class="col-start-1 col-span-2">col 3</div>
    <div class="col-start-1 col-span-2">col 4</div>
    <div class="col-start-1 col-span-4">col 5</div>
  </div>
  ```
  [Tailwind Play example](https://play.tailwindcss.com/2vupBJnRvt)

**Reply by chiu0602** [reply]
Thank you very much, @wongjn! I've already spent several days on this question!

**Reply by chiu0602** [reply]
How about if I want to set this

default (for small screen):
|   |
|---|
|col 1|
|col 2|
|col 3|
|col 4|
|col 5|

for screen larger than md:
|   |   |
|---|---|
|col 1|col 2|
|col 3|col 4|
|col 5|

With following code (need to modify)
```html
<div class="grid-cols-1 md:grid-cols-4 grid">
  <div class="col-span-1">col 1</div>
  <div class="col-span-3">col 2</div>
  <div class="col-span-2">col 3</div>
  <div class="col-span-2">col 4</div>
  <div class="col-span-4">col 5</div>
</div>
```

Thanks

**Reply by wongjn** [reply]
You could consider:
- Removing the grid layout altogether for viewports that don't match the `md` viewport size. The elements would naturally display in one column:
  ```html
  <div class="grid-cols-4 md:grid">
    <div class="col-span-1">col 1</div>
    <div class="col-span-3">col 2</div>
    <div class="col-span-2">col 3</div>
    <div class="col-span-2">col 4</div>
    <div class="col-span-4">col 5</div>
  </div>
  ```
  [Tailwind Play example](https://play.tailwindcss.com/XSTGkxGjxc)
- Applying the `grid-column` properties to the elements elements for viewports that match the `md` viewport size only:
  ```html
  <div class="grid-cols-1 md:grid-cols-4 grid">
    <div class="md:col-span-1">col 1</div>
    <div class="md:col-span-3">col 2</div>
    <div class="md:col-span-2">col 3</div>
    <div class="md:col-span-2">col 4</div>
    <div class="md:col-span-4">col 5</div>
  </div>
  ```
  [Tailwind Play example](https://play.tailwindcss.com/WwOMfNiQ5U)

**Reply by chiu0602** [reply]
Thanks again @wongjn, you saved my day again!
